main/src/cgeo/geocaching/ui/ImageActivityHelper.java

```diff
@@ -39,43 +41,89 @@
  */
 public class ImageActivityHelper {
 
+    //use a globally unique request code to mix-in with Activity.onActivityResult. (This will no longer be neccessary with Activity Result API)
+    //code must be positive (>0) and <=65535 (restriction of SDK21)
+    public static final int REQUEST_CODE_CAMERA = 58221; //this is a random number
+    public static final int REQUEST_CODE_STORAGE_SELECT = 59372;
+    public static final int REQUEST_CODE_STORAGE_SELECT_MULTI = 59373;
+
     private final Activity context;
+    private final Action2<Integer, List<Image>> callbackHandler;
 
-    private final int requestCodeCamera;
-    private final int requestCodeStorageSelect;
-    private final int requestCodeStorageSelectMulti;
+    private final Bundle runningIntents = new Bundle();
 
-    private final Map<Integer, IntentContextData> runningIntents = Collections.synchronizedMap(new HashMap<>());
+    //private final Map<Integer, IntentContextData> runningIntents = Collections.synchronizedMap(new HashMap<>());
```

Comment:
Leftover?

Reply:
Oh man, I am bad... It is good that you find these things! I remove the line and will force-push immediately.